spec/rspec/core/formatters_spec.rb

```diff
@@ -137,6 +138,16 @@ module RSpec::Core::Formatters
           }.to change { loader.formatters.length }
         end
       end
+
+      context "When a custom formatter exists" do
+        before { loader.add Custom::AGeneralFormatter, output }
+
+        it "adds a subclass of that formatter for the same output target" do
+          expect {
+            loader.add Custom::AMoreSpecificFormatter, output
+          }.to change { loader.formatters.length }
```

Comment:
Ah yep, you do need to register otherwise you're considered a legacy formatter, still the new way is better, thanks!